tests/regression/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+        env::var("VALGRIND").is_ok()
+    }
+    //test to create new config, set security policy, host_callback information, load/trust certs, and build config
+    #[test]
+    fn test_set_config() -> Result<(), s2n_tls::error::Error> {
+        if !is_running_under_valgrind() {
+            cg::cachegrind::stop_instrumentation();
+            cg::cachegrind::start_instrumentation();
+            let keypair_rsa = CertKeyPair::default();
+            let _config = set_config(&security::DEFAULT_TLS13, keypair_rsa).expect("Failed to build config");
+            return Ok(());
+        }
+        run_valgrind_test("test_set_config");
+        Ok(())
+    }
+
+    #[test]
+    fn test_rsa_handshake() -> Result<(), s2n_tls::error::Error> {
+        if !is_running_under_valgrind() {
+            cg::cachegrind::stop_instrumentation();
+            // Example usage with RSA keypair (default)
+            let keypair_rsa = CertKeyPair::default();
+            let config = set_config(&security::DEFAULT_TLS13, keypair_rsa)?;
+            // Create a pair (client + server) using that config, start handshake measurement
+            let mut pair = TestPair::from_config(&config);
+            // Assert a successful handshake
+            cg::cachegrind::start_instrumentation();
+            assert!(pair.handshake().is_ok());
+            cg::cachegrind::stop_instrumentation();
+            return Ok(());
+        }
+        run_valgrind_test("test_rsa_handshake");
+        Ok(())
+    }
+    // function to run specified test using valgrind
+    fn run_valgrind_test(test_name: &str) {
+        let exe_path = std::env::args().next().unwrap();
+        let output_file = format!("cachegrind_{}.out", test_name);
+        let valgrind_command = format!(
+            "valgrind --tool=cachegrind --cachegrind-out-file={} {} {}",
+            output_file, exe_path, test_name
+        );
+    
+        println!("Running command: {}", valgrind_command);
+        let output_command = format!("--cachegrind-out-file={}", &output_file);
+        let status = Command::new("valgrind")
+            .args(&["--tool=cachegrind", &output_command, &exe_path, test_name])
+            .env_remove("VALGRIND") //ensures that the recursive call is made to the actual harness code block rather than back to this function
+            .status()
+            .expect("Failed to execute cargo test");
+    
+        if !status.success() {
+            panic!("Valgrind failed");
+        }
+    
+        let annotate_file = format!("perf_outputs/{}.annotated.txt", test_name);
+        // this command annotates the raw output file and pipes it to the specified location
+        let annotate_command = format!(
+            "cg_annotate {} > {}",
+            output_file, annotate_file
+        );
+    
+        println!("Running command: {}", annotate_command);
+    
+        let annotate_status = Command::new("sh")
+            .arg("-c")
+            .arg(&annotate_command)
```

Comment:
Why are we going through a shell here? Generally I would avoid executing shell strings.

Reply:
I'll try using standard I/O, the shell string would directly pipe output into a file but it's not good practice so I'll try I/O